homeassistant/components/sonos/speaker.py

```diff
@@ -193,12 +215,18 @@ def setup(self) -> None:
         else:
             self._platforms_ready.update({BINARY_SENSOR_DOMAIN, SENSOR_DOMAIN})
 
+        self.available_alarms, _ = fetch_alarms_for_speaker(self.soco)
+        if bool(self.available_alarms):
+            dispatcher_send(self.hass, SONOS_CREATE_ALARM, self)
+
+        self._platforms_ready.update({SWITCH_DOMAIN})
```

Comment:
Updating the `_platforms_ready` set is done in `entity.py` here:
https://github.com/home-assistant/core/blob/666c1e44a33edba5e83587a4b2f1e5ed1d7e686b/homeassistant/components/sonos/entity.py#L59-L61

This call will mark it as ready too early.

Reply:
Yes that's True. But we need to add it at least in an else statement, because there might be no alarms for a speaker.

Reply:
Fixed it in new commit.